File: 백준/Bronze/25377. 빵/빵.cs
using System;
using System.IO;

class Program
{
    private const int INF = int.MaxValue;

    static void Main(string[] args)
    {
        new Program().Solution();
    }

    public void Solution()
    {
        int n = int.Parse(Console.ReadLine());
        int min = INF;

        for (int i = 0; i < n; i++)
        {
            string[] input = Console.ReadLine().Split();
            int a = int.Parse(input[0]);
            int b = int.Parse(input[1]);

            if (a > b) continue;
            min = Math.Min(min, b);
        }

        Console.WriteLine(min == INF ? -1 : min);
    }
}

File: 백준/Bronze/25377. 빵/빵.py
import sys
from math import inf

def solution():
    input = sys.stdin.read
    data = input().splitlines()
    
    n = int(data[0])
    min_value = inf
    
    for i in range(1, n + 1):
        a, b = map(int, data[i].split())
        if a > b:
            continue
        min_value = min(min_value, b)
    
    print(-1 if min_value == inf else min_value)

if __name__ == "__main__":
    solution()
